ee/tasks/subscriptions/email_subscriptions.py

```diff
@@ -0,0 +1,73 @@
+import uuid
+from typing import List, Optional
+
+import structlog
+
+from ee.tasks.subscriptions.subscription_utils import UTM_TAGS_BASE
+from posthog.email import EmailMessage
+from posthog.models.exported_asset import ExportedAsset
+from posthog.models.subscription import Subscription, get_unsubscribe_token
+from posthog.utils import absolute_uri
+
+logger = structlog.get_logger(__name__)
+
+
+def send_email_subscription_report(
+    email: str,
+    subscription: Subscription,
+    assets: List[ExportedAsset],
+    invite_message: Optional[str] = None,
+    total_asset_count: Optional[int] = None,
+) -> None:
+    utm_tags = f"{UTM_TAGS_BASE}&utm_medium=email"
+
+    inviter = subscription.created_by
+    is_invite = invite_message is not None
+    self_invite = inviter.email == email
+
+    subject = "Posthog Report"
+    resource_noun = None
+    resource_url = None
+
+    if subscription.insight:
+        resource_name = f"{subscription.insight.name or subscription.insight.derived_name}"
+        resource_noun = "Insight"
+        resource_url = subscription.insight.url
+    elif subscription.dashboard:
+        resource_name = subscription.dashboard.name
+        resource_noun = "Dashboard"
```

Comment:
`noun` and `name` have a lot of overlap. Could this be `resource` and `resource_name`?